servers/zts/src/main/java/com/yahoo/athenz/zts/store/DataStore.java

```diff
@@ -73,6 +70,7 @@
     final Cache<String, PublicKey> zmsPublicKeyCache;
     final Cache<String, List<GroupMember>> groupMemberCache;
     final Cache<String, List<GroupMember>> principalGroupCache;
+    final Map<String, Role> requireRoleCertCache;
```

Comment:
Considered using a principal -> "role tagged as dynamic role cert fetch" cache but this meant that for every change in any role I'll need to go over the entire cache to search for members and fix entries accordingly.
I expect the refresh cert calls / bootstrap calls load will be much lighter so Implemented the following solution:
1.  Created a cache that contains all tagged roles. For every role change we just need to check a specific role and not the entire cache.
2. On certificate refresh / bootstrap we'll go over the relatively small roles list that contains the principal as a member and return it.

Reply:
If we look at the current number of role certificates we issue, it's not a small list. So we need to find a solution to handle these requests efficiently

Reply:
Please see the updated version. I'm keeping cache in 3 data structures:

For regular members - same as for groups using the existing methods
For wildcard members - Set of all roles where a wildcard is a member
For prefix members - Use a prefix trie so finding a match to a given principal is efficient.